controllers/machine.go

```diff
@@ -468,6 +473,102 @@ func byHardwareAffinity(hardware []tinkv1.Hardware, preferred []infrastructurev1
 	}, nil
 }
 
+// ensureHardwareReadyToProvision ensures the hardware is ready to be provisioned.
+// Uses the BMCRef from the hardware to create a BMCJob.
+// The BMCJob is responsible for getting the machine to desired state for provisioning.
+func (mrc *machineReconcileContext) ensureHardwareReadyToProvision(hardware *tinkv1.Hardware) error {
+	if hardware.Spec.BMCRef == nil {
+		mrc.log.Info("Skipping BMC state management", "BMCRef", hardware.Spec.BMCRef, "Hardware", hardware.Name)
+
+		return nil
+	}
+
+	jobName := fmt.Sprintf("%s-provision", mrc.tinkerbellMachine.Name)
+	// Check if BMC Job already exists for the machine
+	exists, err := mrc.bmcJobExists(jobName)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		return nil
+	}
+
+	// Create a BMCJob for hardware provisioning
+	if err := mrc.createBMCJobForHardware(hardware, jobName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// bmcJobExists checks if a BMCJob exists for the machine.
+func (mrc *machineReconcileContext) bmcJobExists(name string) (bool, error) {
+	namespacedName := types.NamespacedName{
+		Name:      name,
+		Namespace: mrc.tinkerbellMachine.Namespace,
+	}
+
+	err := mrc.client.Get(mrc.ctx, namespacedName, &rufiov1.BMCJob{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+
+		return false, fmt.Errorf("checking if bmc job exists: %w", err)
+	}
+
+	return true, nil
+}
+
+// createBMCJobForHardware creates a BMCJob object with the required tasks for hardware provisioning.
+func (mrc *machineReconcileContext) createBMCJobForHardware(hardware *tinkv1.Hardware, name string) error {
+	powerOffAction := rufiov1.HardPowerOff
+	powerOnAction := rufiov1.PowerOn
+	bmcJob := &rufiov1.BMCJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: mrc.tinkerbellMachine.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+					Kind:       "TinkerbellMachine",
+					Name:       mrc.tinkerbellMachine.Name,
+					UID:        mrc.tinkerbellMachine.ObjectMeta.UID,
+				},
+			},
+		},
+		Spec: rufiov1.BMCJobSpec{
+			BaseboardManagementRef: rufiov1.BaseboardManagementRef{
+				Name:      hardware.Spec.BMCRef.Name,
+				Namespace: mrc.tinkerbellMachine.Namespace,
+			},
+			Tasks: []rufiov1.Task{
+				{
+					PowerAction: &powerOffAction,
+				},
+				{
+					OneTimeBootDeviceAction: &rufiov1.OneTimeBootDeviceAction{
+						Devices: []rufiov1.BootDevice{
+							rufiov1.PXE,
+						},
+						EFIBoot: false,
```

Comment:
This should come from the [hardware data](https://github.com/tinkerbell/tink/blob/main/pkg/apis/core/v1alpha1/hardware_types.go#L124).